src/lib/crypto/snarky_tests/snarky_tests.ml

```diff
@@ -599,6 +599,65 @@ module Improper_calls = struct
     ]
 end
 
+(* Tests that check that the hashes of the protocol circuits remain the same *)
+module Protocol_circuits = struct
+  (* Full because we want to be sure nothing changes *)
+  let proof_level, constraint_constants =
+    Genesis_constants.(Proof_level.Full, Constraint_constants.compiled)
+
+  let print_hash print expected digest : unit =
+    if print then Format.printf "expected:\n%s\n\n" expected ;
+    Format.printf "obtained:\n%s\n" digest ;
+    ()
+
+  let blockchain () : unit =
+    let expected = "1f8925ab6b266e96894abecff106a259" in
+
+    let digest =
+      Blockchain_snark.Blockchain_snark_state.constraint_system_digests
+    in
+    let digest = digest ~proof_level ~constraint_constants () in
+    assert (List.length digest = 1) ;
+    let _, hash = List.hd_exn digest in
+    let digest = Md5.to_hex hash in
+
+    let check = String.(digest = expected) in
+    print_hash (not check) expected digest ;
+    assert check ;
+    ()
+
+  let transaction () : unit =
+    let expected1 = "31e96945d5bf7c8d4b1089c59c3b878b" in
+    let expected2 = "0d6b991bd0120736089576f9189ac912" in
+
+    let digest =
+      Transaction_snark.constraint_system_digests ~constraint_constants ()
+    in
+    let hash1, hash2 =
+      match digest with
+      | [ (_, a); (_, b) ] ->
```

Comment:
maybe just `List.(drop (zip digest.(0) digest.(1)) 1)` ? 

But I think the way you have it is better and nicer b/c it errors when there are not `2`

Reply:
that is the idea, if hashes are added or removed it will fail and let us know to update it.